### PR TITLE
Add Doxygen check

### PR DESCRIPTION
**Pull request by AniruddhaKanhere**
Description
-----------
This PR adds check for Doxygen (to check whether the code is properly commented or not).
Few changes in this PR:
- Update `.github/workflows/ci.yml` file to add the check
- Add the config files necessary to run the check in the `docs ` directory.
- Modify the source files to adhere to doxygen standards.

Test Steps
-----------

Related Issue
-----------

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: include/FreeRTOS_IP.h [hunk 1]
```diff
@@ -106,26 +106,32 @@ buffer will have the following contents:
...
 
 	#include "pack_struct_start.h"
+
+	/**
+	* MAC address structure.
+	*/
 	struct xMAC_ADDRESS
 	{
-	uint8_t ucBytes[ ipMAC_ADDRESS_LENGTH_BYTES ];
+	uint8_t ucBytes[ ipMAC_ADDRESS_LENGTH_BYTES ];  /**< Byte array of the MAC address */
```
Comment (yanjos-dev) [inline include/FreeRTOS_IP.h:134]: Missing a period at the end of the comment.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -60,6 +60,19 @@ jobs:
             exit 0
           fi
 
+  doxygen:
+    runs-on: ubuntu-20.04
```
Comment (yanjos-dev) [inline .github/workflows/ci.yml:64]: Does this need to be ubuntu-20.05 instead of ubuntu-latest?

Comment (AniruddhaKanhere) [reply .github/workflows/ci.yml:64]: It does. It cannot work as is with the ubuntu latest. See the yml files in [MQTT](https://github.com/FreeRTOS/coreMQTT/blob/master/.github/workflows/ci.yml#L53-L64) and [HTTP](https://github.com/FreeRTOS/coreHTTP/blob/master/.github/workflows/ci.yml#L53-L65) repo.

**Review by yanjos-dev — APPROVED** [review]

**Comment by AniruddhaKanhere** [comment]
Closing this as a new comprehensive PR shall be opened soon.

**Pull request closed without merging**